### ISSUE TITLE
BUG: If user deletes some items from cart, quoted price does not update to reflect fewer total items

### ISSUE THREAD

**Issue by NullOperator404** [NullOperator404/Developer-Diary#36]

**Comment by NullOperator404** [comment]
The "Total Pieces" and "Order Total" values DO update when a row has been deleted; only the Price Per Piece is not updating to reflect fewer items in the cart.

**Comment by NullOperator404** [comment]
After carefully reviewing the code for a couple of hours, I think I've worked out a solution to this problem. Basically, the Checkout panel will use the `basePrice` value from the first row to refresh the Price Per Piece value. The `trashItem()` function will also have to run a large chunk of the code from `updateNums()` to recalculate and display the new value. Below, I've listed the necessary steps as well as a code except that will need to be implemented into `trashItem()` : 

1. Set `rowId` to "a" (the trash command will default to using the first row to calculate price per piece.) 
2. Get Product # from first row item (see `setRow()` function
3. Use a for loop to iterate through the product database to find the `ProductNumber` of the first-row product. 
4. Set `currentProd` based on result of for loop
5. Use currentProd data to get `BasePrice`
6. Get Total Items in cart value
7. Recalculate price per item and update DOM using the following code: 


```
//UPCHARGE FOR VINYL?
//IF NAMES ONLY
    if(addNames.checked == true && addNumbers.checked == false) {
        vinylCharge = 3.50;
    //IF NUMBERS ONLY
    } else if(addNames.checked == false && addNumbers.checked == true) {
        vinylCharge = 3.50;
    //IF BOTH NAMES AND NUMBERS
    } else if(addNames.checked == true && addNumbers.checked == true) {
        vinylCharge = 7.00;
    } else {
    //IF NO NAMES OR NUMBERS, DON'T UPCHARGE
        vinylCharge = 0;
    }
    //ARE NAMES OR NUMBERS TWO COLOR?
    if(document.getElementById('name-outline').innerHTML !== "None") {
        vinylCharge = vinylCharge + 3.50;
    } else if(document.getElementById('number-outline').innerHTML !== "None") {
        vinylCharge = vinylCharge + 3.50;
    }

    basePrice = basePrice + vinylCharge;

    //CALCULATE BULK DISCOUNT
    if(quantityTotal <= 2) {
        bulkPrice = basePrice;
    } else if(quantityTotal <= 5) {
        bulkPrice = basePrice - 0.45;
    } else if(quantityTotal <= 11) {
        bulkPrice = basePrice - 9.12;
    } else if(quantityTotal <= 23) {
        bulkPrice = basePrice - 13.85;
    } else if(quantityTotal <= 47) {
        bulkPrice = basePrice - 15.92;
    } else if (quantityTotal <= 71) {
        bulkPrice = basePrice - 16.64;
    } else if(quantityTotal <= 99) {
        bulkPrice = basePrice - 17.09;
    } else if(quantityTotal <= 249) {
        bulkPrice = basePrice - 17.95;
    } else if(quantityTotal > 249) {
        bulkPrice = basePrice - 18.49;
    }
    bulkPrice = Math.round(bulkPrice * 100) / 100;
    //ADJUST PRICE FOR NUMBER OF PRINT COLORS
    bulkPrice = bulkPrice + numInks - 1; //MINUS 1 BECAUSE THE FIRST COLOR IS ALREADY ACCOUNTED FOR.

    //UPDATE PRICE PER PIECE
    document.getElementById('basePrice').innerHTML = "$" + (basePrice + numInks);
    document.getElementById('pricePerPiece').innerHTML = "$" + bulkPrice;
    //CALCULATE PERCENTAGE OF BULK DISCOUNT
    discount = Math.round((basePrice - bulkPrice) / (basePrice * 0.01) * 100) / 100;
    document.getElementById('savings').innerHTML = "Your order qualifies for a " + discount + "% bulk discount!";
```

**Comment by NullOperator404** [comment]
9/30/2024

PROBLEM SOLVED! 

A few amendments to my original solution: 

1. It was not necessary to get the `rowId`, as I realized I already had an array setup (`cartProds[]`) that stores the product numbers from the items in the user's cart. Thus, all I needed to do was to pull the data from index 0 of that array, which would be the product number for item 1. 

2. From there, I could proceed to Step 3 in my above comment, only this time the loop through the database is looking for a match to what it found in cartProds[]. Once a match is found, the function updates the global var `currentProd` to the object from the database, and also sets a var called `basePriceTrash` to the `BasePrice` found in the data object. 

3. All that was needed then was to copy/paste the above code from the `updateNums()` function. This block of code handles the calculations to determine the current price per piece. Since the `trashItem()` function had already updated the value of `quantityTotal`, no further code was needed; I only adjusted the names of the variables in the block of code since they aren't global. 

A test resulted in a satisfactory result.
